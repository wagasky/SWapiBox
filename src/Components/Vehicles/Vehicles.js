import React from 'react';
import Card from '../Card/Card';

const Vehicles = ({ data, handleFavorite }) => {

  const renderedCards = data.map( (vehicle, index) => {

    return (
      <Card 
        key={index}
        data={vehicle}
        handleFavorite={handleFavorite}
      />
    );
  });

  return (
    <div className="vehicleCards">
      { renderedCards }
    </div>
  );
};

export default Vehicles;