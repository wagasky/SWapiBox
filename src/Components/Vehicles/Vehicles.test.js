/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import Vehicles from './Vehicles';
import { shallow } from 'enzyme';

describe('Vehicles', () => {

  let renderedComponent;
  let mockHandleFavorite;

  const mockData = [
      {"name":"Sand Crawler","model":"Digger Crawler","vehicleClass":"wheeled","passengers":"30"},
      {"name":"T-16 skyhopper","model":"T-16 skyhopper","vehicleClass":"repulsorcraft","passengers":"1"},
      {"name":"X-34 landspeeder","model":"X-34 landspeeder","vehicleClass":"repulsorcraft","passengers":"1"},
    ]
    
  it('should match snapshot', () => {
    renderedComponent = shallow(<Vehicles data={mockData} handleFavorite={mockHandleFavorite} />);

    expect(renderedComponent).toMatchSnapshot();
  });

});