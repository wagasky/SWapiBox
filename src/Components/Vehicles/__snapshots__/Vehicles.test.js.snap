// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles should match snapshot 1`] = `
<div
  className="vehicle-cards"
>
  <Card
    data={
      Object {
        "model": "Digger Crawler",
        "name": "Sand Crawler",
        "passengers": "30",
        "vehicleClass": "wheeled",
      }
    }
    key="0"
  />
  <Card
    data={
      Object {
        "model": "T-16 skyhopper",
        "name": "T-16 skyhopper",
        "passengers": "1",
        "vehicleClass": "repulsorcraft",
      }
    }
    key="1"
  />
  <Card
    data={
      Object {
        "model": "X-34 landspeeder",
        "name": "X-34 landspeeder",
        "passengers": "1",
        "vehicleClass": "repulsorcraft",
      }
    }
    key="2"
  />
</div>
`;
